enhancements/kube-apiserver/audit-policy.md

```diff
@@ -0,0 +1,494 @@
+      - "/version"
+      - "/healthz"
+      - "/readyz"
+  
+    # Here the tail begins which differs per profile:
+  
+    # A catch-all rule to log all other requests at the Metadata level.
+    - level: Metadata
+      # Long-running requests like watches that fall under this rule will not
+      # generate an audit event in RequestReceived.
+      omitStages:
+      - "RequestReceived"
+   ```
+
+- `OauthUserRequests`: this is like `Default`, but it logs request bodies for requests for oauth users only, i.e. by users outside of the cluster.
+- `NonOpenShiftRequests`: this is like `Default`, but it logs request bodies for requests not coming from known OpenShift components like kubelets, controllers and operators.
+- `AllRequests`: this logs all request bodies (with the exception of the following).
+
+All of the profiles have in common that security sensitive resources, namely
+
+- `secrets`
+- `routes.route.openshift.io`
+
+are never logged beyond metadata level, and
+
+- `oauthaccesstokens.oauth.openshift.io`
+- `oauthauthorizetokens.oauth.openshift.io`
+
+are not logged at all.
+
+Note: this is in line with [etcd-encryption enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/kube-apiserver/encrypting-data-at-datastore-layer.md#proposal)
+with the exception that request level logging includes configmaps.
+
+Any kind of further, deeper policy configuration, e.g. that filters by API groups, user agents or namespaces 
+is explicitly not part of this proposal and requires work on the [upstream dynamic audit](https://github.com/kubernetes/enhancements/blob/f1a799d5f4658ed29797c1fb9ceb7a4d0f538e93/keps/sig-auth/0014-dynamic-audit-configuration.md)
+mechanism and its promotion to beta eventually.
+
+### User Stories [optional]
+
+#### Story 1
+
+As an average customer, I want to have a default audit policy which does not cause
+noticeable resource overhead (cpu and disk).
+
+#### Story 2
+
+As any customer, I don't want to leak sensible information of resources like secrets,
+oauth tokens, etc. in audit logs.
+
+#### Story 3
+
+As a security and regulator demanding customer, I want to log **every** non-sensible resource
+change up to request level detail. 
+
+#### Story 4
+
+As a security and regulator demanding customer, I want to log every non-sensible resource
+change up to request level detail **from non-OpenShift components**.
+
+#### Story 5
+
+As a security and regulator demanding customer, I want to log every non-sensible resource
+change up to request level detail, **from clients outside of the cluster (non-service-accounts)**.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The proposed API looks like this: 
+
+```yaml
+kind: APIServer
+apiVersion: config.openshift.io/v1
+spec:
+  ...
+  audit:
+    profile: Default | OauthUserRequests | NonOpenshiftRequests | AllRequests
+```
+
+In the future, we could add more, even more detailed profiles of logging responses, if customers need this:
```

Comment:
I think this is already too many.  I'd like to see a defense of exclusions from marc above.  They dont' seem to reflect technical realities.